docs/inline_shortcodes.md

```diff
@@ -0,0 +1,119 @@
+
+The complete Wagtail Documentation example can be [Viewed Here](https://docs.wagtail.io/en/stable/extending/extending_draftail.html#creating-new-entities) and this is the first step you should take to get the example working in your own project.
+
+We are making the assumption that in your imported HTML that will be part of a RichText block there is an inline shortcode that will represent a stock to be displayed both in the editor and in the front end.
+
+The word press shortcode will be transformed during the import form  `[stock symbol="TSLA"]` to `<span data-stock="TSLA">$TSLA</span>` before being saved to the RichText block.
+
+You will only need to write one class to handle a shortcode like this. the symbol is dynamic and can be different across multiple shortcode in hte same HTML content.
+
+All shortcodes of the type `[stock symbol="any-valid-symbol"]` will be transformed.
+
+### Stock inline shortcode handler class
+
+```python
+# you can give this module any name or add the class to an existing module
+
+# import the base handler class
+from wagtail_wordpress_import.handle_inline_shortcodes import (
+    InlineShortcodeHandler, 
+)
+
+
+# Subclasses should declare a shortcode_name and provide
+# a construct_html_tag method for converting the shortcode to a HTML tag.
+# @register()
+class StockHandler(InlineShortcodeHandler):
+    """
+    The Wordpress shortcode is replaced by a custom HTML tag. 
+    The shortcode attributes are preserved and can be included in the custom HTML tag.
+
+    Sample wordpress stock shortcode:
+    [stock symbol="TSLA"]
+
+    all matching shortcodes are replaced by:
+
+    <span data-stock="TSLA">$TSLA</span>
+    """
+
+    # The shortcode name that is after the first "[" and the 
+    # matching for the shortcode name will end at the first space
+    shortcode_name = "stock"
+
+    """You must implement this method in your own class"""
+    def construct_html_tag(self, html):
+        # It will receive the `html` string content of the RichText block
+
+        """ This is fully functioning code for the stock example. You may need 
+        to use your own logic here depending on your requirements.
+        
+        The output needs be the `html` string received with the  modifications 
+        your code makes.
+        """
+
+        matches = self._pattern.finditer(html) # find all matches of the shortcode
+
+        for match in matches: # Loop through the matches
+
+            # Get the shortcode attributes using the parent class method
+            # or you can implement your own method for this.
+            attrs = self.get_shortcode_attrs(match.groupdict()["attrs"])
+
+            # Modify the `html` string by replacing the shortcode with a HTML tag.
+            html = html.replace(
+                match.group(),
+                f'<{self.element_name} data-{self.shortcode_name}="{attrs["symbol"]}">${attrs["symbol"]}</span>',
```

Comment:
Self review: i noticed this should be:

f'<{self.element_name} data-{self.shortcode_name}="{attrs["symbol"]}">${attrs["symbol"]}</{self.element_name}>',
If you set element_name in you class